src/ast/passes/semantic_analyser.cpp

```diff
@@ -1176,7 +1176,7 @@ void SemanticAnalyser::visit(Call &call)
           << "BPF_FUNC_d_path not available for your kernel version";
     }
 
-    if (check_varargs(call, 1, 1)) {
+    if (check_varargs(call, 1, 2)) {
```

Comment:
Let's add some additional type checking for the second arg if it exists e.g.
https://github.com/bpftrace/bpftrace/blob/master/src/ast/passes/semantic_analyser.cpp#L1212

Reply:
Also line 1194 probably needs to be adjusted to take this second arg if it exists:
```
call.type = SizedType(Type::string,
                            bpftrace_.config_.get(ConfigKeyInt::max_strlen));
                            ```

Reply:
type checking is added.

`call.type` is used for the return type of the call, am I right?

Reply:
Yeah exactly.